File: py/1094.py
from sys import stdin

cases = []
count = int(stdin.readline())
while count > 0:
    cases.append(list(stdin.readline().split()))
    count -= 1

rabbits = 0
rats = 0
frogs = 0
for c in cases:
    c[0] = int(c[0])
    
    if c[-1] == 'C':
        rabbits += c[0]
    elif c[-1] == 'R':
        rats += c[0]
    else:
        frogs += c[0]
        
total = rabbits + rats + frogs

print(f'Total: {total} cobaias')
print(f'Total de coelhos: {rabbits}')
print(f'Total de ratos: {rats}')
print(f'Total de sapos: {frogs}')
print(f'Percentual de coelhos: {(rabbits / total * 100):.2f} %')
print(f'Percentual de ratos: {(rats / total * 100):.2f} %')
print(f'Percentual de sapos: {(frogs / total * 100):.2f} %')

File: cpp/1094.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    int rabbits = 0, rats = 0, frogs = 0, total, amount, n;
    char animal;
    cin >> n;

    for (n; n > 0; n--) {
        cin >> amount >> animal;

        switch (animal)
        {
            case 'C':
                rabbits += amount;
                break;
            
            case 'R':
                rats += amount;
                break;

            case 'S':
                frogs += amount;
                break;
        }
    }

    total = rabbits + rats + frogs;
    printf("Total: %d cobaias\n", total);
    printf("Total de coelhos: %d\n", rabbits);
    printf("Total de ratos: %d\n", rats);
    printf("Total de sapos: %d\n", frogs);
    printf("Percentual de coelhos: %.2f %\n", (float)(rabbits) / total * 100);
    printf("Percentual de ratos: %.2f %\n", (float)(rats) / total * 100);
    printf("Percentual de sapos: %.2f %\n", (float)(frogs) / total * 100);
    return 0;
}